src/main/java/com/mucommander/main/AutoProcessor.java

```diff
@@ -141,52 +164,62 @@ private static void processAutoDeploy(Map configMap, BundleContext context)
             // Look in the specified bundle directory to create a list
             // of all JAR files to install.
             File[] files = new File(autoDir).listFiles();
-            List jarList = new ArrayList();
+            List<File> jarList = new ArrayList<>();
             if (files != null)
             {
                 Arrays.sort(files);
-                for (int i = 0; i < files.length; i++)
+                for (File f : files)
                 {
-                    if (files[i].getName().endsWith(".jar"))
+                    if (f.getName().endsWith(".jar"))
                     {
-                        jarList.add(files[i]);
+                        jarList.add(f);
                     }
                 }
             }
 
             // Install bundle JAR files and remember the bundle objects.
-            final List startBundleList = new ArrayList();
-            for (int i = 0; i < jarList.size(); i++)
+            final List<Future<Bundle>> startBundleList = new ArrayList<>();
+            final List<Future<Bundle>> deferStartBundleList = new ArrayList<>();
+            for (File jarFile : jarList)
             {
                 // Look up the bundle by location, removing it from
                 // the map of installed bundles so the remaining bundles
                 // indicate which bundles may need to be uninstalled.
-                Bundle b = (Bundle) installedBundleMap.remove(
-                    ((File) jarList.get(i)).toURI().toString());
-
+                Bundle b = (Bundle) installedBundleMap.remove(jarFile.toURI().toString());
+                Future<Bundle> futureBundle = CompletableFuture.completedFuture(b);
+                var defer = false;
                 try
                 {
                     // If the bundle is not already installed, then install it
                     // if the 'install' action is present.
                     if ((b == null) && actionList.contains(AUTO_DEPLOY_INSTALL_VALUE))
                     {
-                        b = context.installBundle(
-                            ((File) jarList.get(i)).toURI().toString());
+                        var jarPath = jarFile.toURI().toString();
+                        if (jarPath.contains("dropbox") || jarPath.contains("jediterm")
+                                || jarPath.contains("protocol-gcs") || jarPath.contains("viewer-pdf")
+                                || jarPath.contains("gdrive")) {
```

Comment:
that's not that great.. what's the reason for this?

Reply:
yes, it's not, I agree. Those bundles are relatively big and not required as the first ones. We may introduce sth like checking bundle size, and defer the biggest ones.

Reply:
I'm ok with using several threads to load the bundles but we need to ensure that the protocol and format bundles are loaded before mucommander-core is loaded as we may have an initial location that requires them

Reply:
I may try to drop this deferring and see how it impacts the load time.

Reply:
> I may try to drop this deferring and see how it impacts the load time.

in my tests, I didn't see an issue with deferring `gdrive` even when I had my google drive loaded on startup but @pskowronek did you check the above?